stdlib/system/src/main/ballerina/src/system/process.bal

```diff
@@ -0,0 +1,125 @@
+    public function exitCode() returns int|Error {
+        return nativeExitCode(self);
+    }
+    
+    # Destroys the process.
+    public function destroy() {
+        return nativeDestroy(self);
+    }
+    
+    # Provides a channel to write into, where this data is made available as
+    # the standard input for the process.
+    #
+    # + return - The `io:WritableByteChannel` representing the channel to write into for process's standard input
+    public function stdin() returns io:WritableByteChannel {
+        return nativeStdin(self);
+    }
+    
+    # Provides a channel to read from, where this data as made available from
+    # the standard output of the process.
+    #
+    # + return - The `io:ReadableByteChannel` representing the channel to read representing process's standard output
+    public function stdout() returns io:ReadableByteChannel {
+        return nativeStdout(self);
+    }
+    
+    # Provides a channel to read from, where this data as made available from
+    # the standard error of the process.
+    #
+    # + return - The `io:ReadableByteChannel` representing the channel to read representing process's standard error
+    public function stderr() returns io:ReadableByteChannel {
+        return nativeStderr(self);
+    }
+    
+    # Pipes the standard output of the current process to the standard input of the given process.
+    #
+    # + process - The process to pipe the data to
+    # + return - The process that is passed in, where this is used to help chain pipe operations
+    public function pipe(Process process) returns Process {
+        _ = start self.doPipe(self.stdout(), process.stdin());
+        return process;
+    }
+    
+    private function doPipe(io:ReadableByteChannel input, io:WritableByteChannel output) {
+        while (true) {
+            [byte[], int]|error result = input.read(self.BUF_SIZE);
+            if (result is error) {
+                io:println("Error in pipe read: ", result);
+                break;
+            } else {
+                if (result[1] <= 0) {
+                    break;
+                } else {
+                    int i = 0;
+                    while (i < result[1]) {
+                        var result2 = output.write(result[0], i);
+                        if (result2 is error) {
+                            io:println("Error in pipe write: ", result2);
+                            break;
+                        } else {
+                            i = i + result2;
+                        }
+                    }
+                }
+            }
+        }
+        
+        var cr2 = input.close();
+        if (cr2 is error) {
+            io:println("Error closing pipe input: ", cr2);
+        }
+        var cr1 = output.close();
+        if (cr1 is error) {
+            io:println("Error closing pipe output: ", cr1);
+        }
+        
+    }
+
+};
+
+function nativeWaitForExit(Process process) returns int|Error = external;
```

Comment:
We can have this function inside the object scope. So we can omit the passing process object as a parameter. Basically now, we are no longer following this approach.

Reply:
Thank you, and fixed.